src/Chapter05.Tests/Listing05.31.Tests.cs

```diff
@@ -5,4 +5,22 @@ namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter05.Listing05_31.Tests;
 [TestClass]
 public class ProgramTests
 {
+    [TestMethod]
+    public void Main_NoArgs_ExpectErrors()
+    {
+        string expected = "Value cannot be null. (Parameter \'httpsUrl\')";
+
+        IntelliTect.TestTools.Console.ConsoleAssert.Expect(expected,
+        () =>
+        {
+            try
+            {
+                Program.Main();
+            }catch(Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
```

Comment:
Won't it not write to console causing it to fail?

Reply:
O I get what you are doing now. Yes this will work, is just doing string comparisons so different Exception types might not be the same, but I am fine with this. Disregard my prior comments, this does work. I have an issue logged on the ConsoleAssert class to make a real method to assert on exceptions.